equinox/nn/pool.py

```diff
@@ -0,0 +1,245 @@
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.random
+import numpy as np
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+class Pool(Module):
+    """General N-dimensional downsampling over a sliding window."""
+
+    operation: Callable[[Array, Array], Array]
+    num_spatial_dims: int = static_field()
+    kernel_size: Union[int, Sequence[int]] = static_field()
+    stride: Union[int, Sequence[int]] = static_field()
+    padding: Union[int, Sequence[int]] = static_field()
+    init: float = static_field()
+
+    def __init__(
+        self,
+        init: int,
```

Comment:
Nit: not necessarily an integer! e.g. `jnp.NINF` is a float.

Admittedly I'm not sure what the correct annotation is here. `Union[int, float]` maybe? Probably `Array`s are valid too?